### DISCUSSION TITLE
Form Data when control moves to another component

### DISCUSSION THREAD

**Discussion by sivaputrevu** [react-hook-form/react-hook-form#3276] [General]
Hi,

I need one info on to hold the form data. when the user clicks on submit, depends on his selection I need to show a confirmation dialog[which is entirely a diff component] and the user confirms the form data will get submitted to server. for this req. as of now I am caching the form data in a local state and fetching back when the user confirmed in the confirmation dialog. Is there any way to hold on the form data[React hook form way] when the control moves to another component and comes back!

please assist!

**Comment by bluebill1049** [comment]
Is this what you after? https://react-hook-form.com/advanced-usage#WizardFormFunnel

**Reply by sivaputrevu** [reply]
cool! sometime back I have seen the same example. you have used a state mgmt library to track the user data across multiple components If I am not wrong? Is there any other way of doing it without using any state mgmt library?

**Reply by bluebill1049** [reply]
you can use `useState` or `useContext` to hoist your state into a store. it works the same.
